### ISSUE TITLE
Test bundling doesn't work on physical iOS devices

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#1297]
Since iOS 14, Flutter apps built in debug/profile mode must be run through a debugger.

Before test bundling, we've been accomplishing this by doing just this – starting the app under test in debug mode (with the lldb debugger attached using `ios-deploy`).

This approach breaks with test bundling because we're running every Dart test in a separate app process – which doesn't work, because there's no debugger attached.

In other words, this is prohibited:

```objective-c
[[[XCUIApplication alloc] init] launch];
```

We depend on the above heavily.

The only workaround I see around this is to build the app in release mode.

**Comment by bartekpacia** [comment]
Trying to run tests in release-built app, but failing to do so.

Possible cause: `RunnerTests` is present in the `.xctestrun`, but we want only `RunnerUITests`

**Comment by bartekpacia** [comment]
More notes about the error above: the test runner process dies before `testInvocations` is called. Verified by:

```objective-c
  +(NSArray<NSInvocation *> *)testInvocations {                                                                 
    NSLog(@"testInvocations called");
    // ...
  }
```

That line is not visible in the logs.

**Comment by sonnyvesali** [comment]
@bartekpacia how should the problem be addressed, I'm currently following the Native automation, and the android side of things is working fine, but the iOS side has some funky behavior where it attempts to enter the RunnerUITests, bundle but it goes into the bundle, a black screen that says `Test Starting` or something like that is shown, the app rotates and then crashes and then in terminal it says the test passes.

Logs:
```
No device specified, using the first one (iPhone 14 Pro)
Every test target will be run 1 time(s)
• Building app with entrypoint example_test.dart for iOS simulator (debug)...
        2023-05-21 15:22:30.199 xcodebuild[74409:1090411] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
✓ Completed building app with entrypoint example_test.dart for iOS simulator (92.5s)
• Running app with entrypoint example_test.dart for iOS simulator on simulator iPhone 14 Pro...
        2023-05-21 15:23:48.321 xcodebuild[80893:1108554] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
        --- xcodebuild: WARNING: Using the first of multiple matching destinations:
        { platform:iOS Simulator, id:4D42B44D-F2D5-4ED8-927C-D2FA002AE3DF, OS:16.4, name:iPhone 14 Pro }
        { platform:iOS Simulator, id:4D42B44D-F2D5-4ED8-927C-D2FA002AE3DF, OS:16.4, name:iPhone 14 Pro }
        2023-05-21 15:24:39.433 xcodebuild[80893:1108554] [MT] IDETestOperationsObserverDebug: 50.922 elapsed -- Testing started completed.
        2023-05-21 15:24:39.433 xcodebuild[80893:1108554] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
        2023-05-21 15:24:39.433 xcodebuild[80893:1108554] [MT] IDETestOperationsObserverDebug: 50.923 sec, +50.923 sec -- end
        2023-05-21 15:24:39.691 xcodebuild[80893:1108554] [MT] IDEResultKit: Warning: While writing a result bundle to /Users/sonny/Library/Developer/Xcode/DerivedData/Runner-hjbpzaxewbdgumfnlwmabuyumbsv/Logs/Test/Test-Runner-2023.05.21_15-23-48--0500.xcresult, not all contents have been imported, as the Staging directory still contains the following files/directories: ["1_Test"]
✓ Completed executing app with entrypoint example_test.dart for iOS simulator on simulator iPhone 14 Pro (57.0s)
```

**Comment by bartekpacia** [comment]
This issue is about a feature that isn't yet released to the stable channel.

If you're experiencing problems with the stable version, please report a new bug.
